packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -6,12 +6,29 @@
  */
 
 import * as vscode from 'vscode';
+import { LanguageClient } from 'vscode-languageclient';
+import { DEBUGGER_LINE_BREAKPOINTS } from './constants';
 import * as languageServer from './languageServer';
 
+let languageClient: LanguageClient | undefined;
+
 export async function activate(context: vscode.ExtensionContext) {
-  const languageClient = await languageServer.createLanguageServer(context);
+  languageClient = await languageServer.createLanguageServer(context);
   const handle = languageClient.start();
   context.subscriptions.push(handle);
+
+  const exportedApi = {
+    getLineBreakpointInfo
+  };
+  return exportedApi;
+}
+
+async function getLineBreakpointInfo(): Promise<{}> {
+  let response = {};
+  if (languageClient) {
+    response = await languageClient.sendRequest(DEBUGGER_LINE_BREAKPOINTS);
```

Comment:
This is the part I'm not too sure about: what happens if the languageClient restarts? Is `languageClient` still valid in that case?

There are cases where the language client will restart. I am not entirely sure when it happens. But does that mean that you might have an invalid exported session?